src/System.Runtime.Extensions/tests/System/IO/PathTests.cs

```diff
@@ -203,14 +213,21 @@ public static void GetFileName(string path, string expected)
         }
 
         [PlatformSpecific(TestPlatforms.AnyUnix)]  // Tests Unix-specific valid file names
-        [Fact]
-        public static void GetFileName_Unix()
+        [Theory]
+        [InlineData(" . ")]
+        [InlineData(" .. ")]
+        [InlineData("fi le")]
+        [InlineData("fi  le")]
+        public static void GetFileName_Unix(string file)
         {
-            Assert.Equal(" . ", Path.GetFileName(" . "));
-            Assert.Equal(" .. ", Path.GetFileName(" .. "));
-            Assert.Equal("fi le", Path.GetFileName("fi le"));
-            Assert.Equal("fi  le", Path.GetFileName("fi  le"));
-            Assert.Equal("fi  le", Path.GetFileName(Path.Combine("b \r\n ar", "fi  le")));
+            Assert.Equal(file, Path.GetFileName(file));
+            Assert.Equal(file, new string(Path.GetFileName(file).AsReadOnlySpan()));
+
+            if (file.Equals("fi  le"))
+            {
+                Assert.Equal(file, Path.GetFileName(Path.Combine("b \r\n ar", file)));
+                Assert.Equal(file, new string(Path.GetFileName(Path.Combine("b \r\n ar", file).AsReadOnlySpan())));
+            }      
```

Comment:
It's strange that this section is included in this test.  It applies to only one of the InlineDatas, and thus isn't actually parameterized based on the input.  It should be moved to its own test.